sh_scrapy/settings.py

```diff
@@ -1,40 +1,51 @@
+import logging
 import sys, os, tempfile
 from sh_scrapy.compat import to_native_str, is_string
+from scrapy.settings import BaseSettings
+from scrapy.utils.misc import load_object
 from scrapy.utils.project import get_project_settings
 
 
+logger = logging.getLogger(__name__)
 REPLACE_ADDONS_PATHS = {
     "hworker.bot.ext.page.PageStorageMiddleware":
         "scrapy_pagestorage.PageStorageMiddleware",
     "hworker.bot.ext.persistence.DotScrapyPersistence":
         "scrapy_dotpersistence.DotScrapyPersistence",
 }
+SLYBOT_SPIDER_MANAGER = 'slybot.spidermanager.ZipfileSlybotSpiderManager'
 
 try:
     from scrapy.utils.deprecate import update_classpath
 except ImportError:
     update_classpath = lambda x: x
 
 
-def _update_settings(o, d):
-    # We need to convert settings to string since the S3 download handler
-    # doesn't work if the AWS keys are passed as unicode. Other code may also
-    # depend on settings being str. TODO: we should test this
-    for k, v in d.items():
-        d[to_native_str(k)] = to_native_str(v) if is_string(v) else v
-    o.update(d)
+def _update_settings(o, changes, priority='default'):
+    """
+    We need to convert settings to string since the S3 download handler
+    doesn't work if the AWS keys are passed as unicode. Other code may also
+    depend on settings being str.
 
+    :param o: BaseSettings object to merge settings provided by d
```

Comment:
and btw, why not to go a step further and not to subclass BaseSettings? e.g. override `set` method to convert keys and, optionally, values to native string.